### PR TITLE
[test optimization] Add Dynamic Instrumentation to mocha retries 

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?
Add support for dynamic instrumentation for mocha.

### Motivation
Add Dynamic instrumentation to mocha retries

### Plugin Checklist

- [x] Unit tests.

### CODE REVIEW

File: integration-tests/jest/jest.spec.js [hunk 1]
```diff
@@ -2552,7 +2552,7 @@ describe('jest CommonJS', () => {
         itr_enabled: false,
         code_coverage: false,
         tests_skipping: false,
-        flaky_test_retries_enabled: true,
+        flaky_test_retries_enabled: false,
         early_flake_detection: {
           enabled: false
         }
```
Comment (juan-fernandez) [inline integration-tests/jest/jest.spec.js:2558]: unrelated fixes: we don't want to connect with auto test retries yet

**Review by anmarchenko — APPROVED** [review]
